examples/02_intermediate/wikimedia.ipynb

```diff
@@ -194,39 +200,53 @@
+    "                        timestamp = change['meta']['dt']\n",
+    "                        event = f\"{timestamp}:: {change['user']} edited {change['title']}\"\n",
     "                        servername = change['server_name']\n",
     "                        if servername in servernames:\n",
     "                            servernames[servername] += 1\n",
     "                        else:\n",
     "                            servernames[servername] = 1\n",
-    "                        # Update perspective widget with this event\n",
-    "                        self.widget.update(\n",
-    "                            {\n",
-    "                                \"servername\": [change['server_name']],\n",
-    "                                \"servername_count\": [servernames[servername]],\n",
-    "                                \"time\": [change['meta']['dt']],\n",
-    "                                \"event\": [f\"{change['user']} edited {change['title']}\"],\n",
-    "                            }\n",
-    "                        )\n",
     "                        # Manually \"tick\" this edge\n",
-    "                        self.push_tick(event)\n",
+    "                        self.push_tick(\n",
+    "                            WikiData(\n",
+    "                                servername=servername,\n",
+    "                                servername_count=servernames[servername],\n",
+    "                                timestamp=timestamp,\n",
+    "                                event=event\n",
+    "                            )\n",
+    "                        )\n",
+    "\n",
+    "# Create the graph-time representation of our adapter\n",
+    "FetchWikiData = py_push_adapter_def(\"FetchWikiData\", FetchWikiDataAdapter, csp.ts[WikiData])\n",
+    "\n",
+    "@csp.node\n",
+    "def update_widget(wiki_event: csp.ts[WikiData], widget: PerspectiveWidget) -> csp.ts[str]:\n",
+    "    widget.update({\n",
+    "        \"servername\": [wiki_event.servername],\n",
+    "        \"servername_count\": [wiki_event.servername_count],\n",
+    "        \"timestamp\": [wiki_event.timestamp],\n",
+    "        \"event\": [wiki_event.event],\n",
+    "    })\n",
+    "    return wiki_event.event\n",
     "\n",
     "@csp.graph\n",
     "def wiki_graph():\n",
     "    print(\"Start of graph building\")\n",
-    "    # Create the graph-time representation of our adapter\n",
-    "    FetchWikiData = py_push_adapter_def(\"FetchWikiData\", FetchWikiDataAdapter, csp.ts[Event], widget=PerspectiveWidget)\n",
     "    # Create an edge in the graph with the events fetched by the adapter\n",
-    "    event = FetchWikiData(widget=widget)\n",
+    "    result = FetchWikiData()\n",
+    "    # Update Perspective widget with each event/tick\n",
+    "    event = update_widget(result, widget=widget)\n",
+    "    csp.print(\"Event\", event)\n",
```

Comment:
FYI you can actually pass result.event here, CSP will pull the struct field out into its own time series